src/librustc_trans/back/write.rs

```diff
@@ -512,6 +517,13 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
         None => {},
     }
 
+    // Inject a local-scoped `_Unwind_Resume` function, which redirects to `rust_eh_unwind_resume`.
+    // This must be done after optimizations, because until DwarfEHPrepare pass has been run,
+    // `_Unwind_Resume` is not referenced by any live code and would have been eliminated by opt.
```

Comment:
This may also be able to prevent elimination through the [`llvm.used` global](http://llvm.org/docs/LangRef.html#the-llvm-used-global-variable)